### PR TITLE
Chain observer retrieves the current Chain Point

### PR DESCRIPTION
**Pull request by jpraynaud**
## Content
This PR includes the support for the chain observers of the retrieval of the current Cardano chain point:
- Upgraded `ChainObserver` trait.
- Upgraded `FakeChainObserver` and mocks implementations.
- Upgraded `CardanoCliObserver` implementation.
- Prepared `PallasChainObserver` implementation (not implemented yet).

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

## Comments
The `PallasChainObserver` does not currently implement the `get_current_chain_point`. 
This will be done in a separate PR by @falcucci .

## Issue(s)
Relates to #1589

### CODE REVIEW

**Review by dlachaume — APPROVED** [review]
LGTM

File: mithril-common/src/chain_observer/pallas_observer.rs [hunk 1]
```diff
@@ -384,6 +384,11 @@ impl ChainObserver for PallasChainObserver {
         Ok(Some(Epoch(epoch as u64)))
     }
 
+    async fn get_current_chain_point(&self) -> Result<Option<ChainPoint>, ChainObserverError> {
+        // TODO: Implement get_current_chain_point with pallas
```
[the remark this answers is not in the document]
Comment (jpraynaud) [reply mithril-common/src/chain_observer/pallas_observer.rs:388]: Yes, this part will be implemented by @falcucci  :+1:

File: mithril-common/src/chain_observer/cli_observer.rs [hunk 2]
```diff
@@ -485,6 +532,22 @@ mod tests {
...
     }
 
+    #[tokio::test]
+    async fn test_get_current_chain_point() {
+        let observer = CardanoCliChainObserver::new(Box::<TestCliRunner>::default());
+        let chain_point = observer.get_current_chain_point().await.unwrap().unwrap();
+
+        assert_eq!(
+            ChainPoint {
+                slot_number: 25886617,
```
[the remark this answers is not in the document]
Comment (jpraynaud) [reply mithril-common/src/chain_observer/cli_observer.rs:542]: They are coming from the test CLI runner:
https://github.com/input-output-hk/mithril/blob/6ecac83390b20f89dd521d19e4560d60a2b882ac/mithril-common/src/chain_observer/test_cli_runner.rs#L173

File: mithril-common/src/entities/cardano_transaction.rs [hunk 3]
```diff
@@ -1,13 +1,12 @@
 use crate::crypto_helper::MKTreeNode;
 
-use super::{BlockNumber, ImmutableFileNumber};
+use super::{
+    cardano_chain_point::{BlockHash, SlotNumber},
+    BlockNumber, ImmutableFileNumber,
+};
```
Comment (Alenar) [inline mithril-common/src/entities/cardano_transaction.rs:6]: We should avoid `use super::` in production code, (it's fine for tests). can you replace it with a `use crate::entities::` instead ?

File: mithril-common/src/entities/block_range.rs [hunk 4]
```diff
@@ -11,8 +11,7 @@ use crate::{
     StdResult,
 };
 
-/// BlockNumber is the block number of a Cardano transaction.
-pub type BlockNumber = u64;
+use super::BlockNumber;
```
Comment (Alenar) [inline mithril-common/src/entities/block_range.rs:14]: We should avoid `use super::` in production code, (it's fine for tests). can you replace it with a `use crate::entities::` instead ?

**Review by Alenar — APPROVED** [review]
LGTM, just two minor comments :rocket:.

File: mithril-common/src/entities/cardano_transaction.rs (hunk 3 above)
Comment (jpraynaud) [reply mithril-common/src/entities/cardano_transaction.rs:6]: Done

File: mithril-common/src/entities/block_range.rs (hunk 4 above)
Comment (jpraynaud) [reply mithril-common/src/entities/block_range.rs:14]: Done

**Pull request merged**